OSDOCS#9222: Agent installer on multi-arch (#69883)
openshift/openshift-docs: modules/agent-installer-architectures.adoc

// Module included in the following assemblies:
//
// * installing/installing-with-agent-based-installer/preparing-to-install-with-agent-based-installer.adoc

:_mod-docs-content-type: PROCEDURE
[id="agent-install-networking_{context}"]
= Optional: Verifying architectures on the Agent-based installer 

You can check which architecture your {product-title} agent installer cluster is using by viewing the version binary.

.Prerequisites: 

* You installed the OpenShift CLI (oc)

> 

Reviewer: * you have the openshift-installer

.Procedure: 

> 

Reviewer: Get the release payload from the openshift-installer:
$ ./openshift-install version
then determine if the release payload supports multiple architectures:

$ oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:27d41f99e0784a2f596e773799983147b794df48d1f7536d51a95a60945b1931 -o jsonpath="{ .metadata.metadata}"

If the above returns {"release.openshift.io/architecture":"multi"}, then this installer will support arm, power, s390x, and x86.

Otherwise you can only install the architecture defined for "release architecture" outputted from  `openshift-install version`

// Module included in the following assemblies:
//
// * installing/installing-with-agent-based-installer/preparing-to-install-with-agent-based-installer.adoc

:_mod-docs-content-type: PROCEDURE
[id="agent-install-networking_{context}"]
= Optional: Verifying architectures on the Agent-based installer 

You can check which architecture your {product-title} agent installer cluster is using by viewing the version binary.

.Prerequisites: 

* You installed the OpenShift CLI (oc)
* You have the openshift-installer binary

.Procedure: 

. Check your release payload by running the following command: 
[source,terminal]
+
----
$ ./openshift-install version
----
+
.Example output 
[source,terminal]
----

> ./openshift-install 4.15.0-ec.3

Reviewer: missed this on my first pass.. this should be `openshift-install version`

Author: Oh wait, I don't think the command needs to be in the example output.

Reviewer: ok.. i'm being dumb.. that is part of the output 🤦 
cool if we make it generic? something like
`./openshfit-install 4.15.z` or something?

built from commit e80d50ac8c9ab65833045d648a01eb1a0a3a6d7d
release image quay.io/openshift-release-dev/ocp-release@sha256:27d41f99e0784a2f596e773799983147b794df48d1f7536d51a95a60945b1931

> release architecture amd64

Reviewer: nit, can we make up a commit and release sha? feels weird to have real values here?

Author: Yes, looking at that now!

----
+
If you are using the `amd64` or `multi` installer payload, the `release architecture` output of this command displays to `amd64` by default

. To verify that you are using your preferred architecture, run the following command:
[source,terminal]
+
----
$ oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:27d41f99e0784a2f596e773799983147b794df48d1f7536d51a95a60945b1931 -o jsonpath="{ .metadata.metadata}"
----
+
.Example output of a cluster using the `multi` binary
[source,terminal]
----
release@sha256:27d41f99e0784a2f596e773799983147b794df48d1f7536d51a95a60945b1931 -o jsonpath="{ .metadata.metadata}"
{"release.openshift.io/architecture":"multi"}
----
+

> With the `multi` installer binary, your cluster supports `arm64`, `amd64`, `s390x`, and `ppc64le`. Otherwise, you can only install the architecture defined in the `release architecture` output from the `./openshift-install version` command. 

Reviewer: We should wait and see if the enablement work for s390x and ppc64le make 4.15, assuming they do this should be fine.  cc: @jaypoulz

Reviewer 2: According to our testing, we're looking good for ppc64le and s390x for 4.15.

Would be good to get QE to put the the next multi payload through it's paces for those arches tho.

// Module included in the following assemblies:
//
// * installing/installing-with-agent-based-installer/preparing-to-install-with-agent-based-installer.adoc

:_mod-docs-content-type: PROCEDURE
[id="agent-install-networking_{context}"]
= Optional: Verifying architectures on the Agent-based installer 

You can check which architecture your {product-title} agent installer cluster is using by viewing the version binary.

.Prerequisites: 

* You installed the OpenShift CLI (oc)
* You have the openshift-installer binary

.Procedure: 

. Check your release payload by running the following command: 
[source,terminal]
+
----
$ ./openshift-install version
----
+
.Example output 
[source,terminal]
----
./openshfit-install 4.15.z

> built from commit e80d50ac8c9ab65833045d648a01eb1a0a3a6d7d

Author: @jeffdyoung So since this is an example output, I don't think its necessary to put a open ended value here. Usually those updatable values (usually formated like this <image>) are for commands, so that the user can update them. 
I did update it on [line 40 ](https://github.com/openshift/openshift-docs/pull/69883/files#diff-01f0b2e4a4420fd86c18783d418edd6deb180ba575271bfd5a21cec9469f3f43R40) so its still like a value.

// Module included in the following assemblies:
//
// * installing/installing-with-agent-based-installer/preparing-to-install-with-agent-based-installer.adoc

:_mod-docs-content-type: PROCEDURE
[id="agent-install-networking_{context}"]
= Optional: Verifying architectures on the Agent-based installer 

You can check which architecture your {product-title} agent installer cluster is using by viewing the version binary.

.Prerequisites: 

* You installed the OpenShift CLI (oc)
* You have the openshift-installer binary

.Procedure: 

. Check your release payload by running the following command: 
[source,terminal]
+
----

> $ ./openshift-install version

Reviewer 3: openshift-install version

> $ ./openshift-install version

Reviewer 3: ```suggestion
$ openshift-install version
```

----
+
.Example output 
[source,terminal]
----

> ./openshfit-install 4.15.z

Reviewer 3: ```suggestion
openshift-install version
bin/openshift-install 4.15.0-rc.2
built from commit c20bcb355061fc9055992b18cca08683511b676c
release image registry.ci.openshift.org/ocp/release@sha256:baed442e1faf17a500b0f0952ecbcfad66e03074c33fb4f08fa95f49bd8fc4ff
release architecture amd64
```

built from commit e80d50ac8c9ab65833045d648a01eb1a0a3a6d7d
release image quay.io/openshift-release-dev/ocp-release@sha256:27d41f99e0784a2f596e773799983147b794df48d1f7536d51a95a60945b1931
release architecture amd64
----
+

> If you are using the `amd64` or `multi` installer payload, the `release architecture` output of this command displays to `amd64` by default

Reviewer 4: ```suggestion
If you are using the `multi` installer payload, the `release architecture` output of this command displays to the architecture of the host you are running the installer in.
```

Author: I changed it just a touch to make it more direct and avoid the preposition at the end

. To verify that you are using your preferred architecture, run the following example command:
[source,terminal]
+
----
$ oc adm release info <release-image> -o jsonpath="{ .metadata.metadata}"
----
+
[source,terminal]
----

> $ oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:27d41f99e0784a2f596e773799983147b794df48d1f7536d51a95a60945b1931 -o jsonpath="{ .metadata.metadata}"

Reviewer 3: ```suggestion
```

Author: Hello! Not sure what part you wanted to update here cuz I think that is the command that is being used.

----
+

> You can access your release image from the ./openshift-install version command

Reviewer 3: ```suggestion
You can access your release image from the 'openshift-install version' command
```

> You can access your release image from the ./openshift-install version command

Reviewer 3: ```suggestion
```

Author: Same with this part

+

> .Example output of a cluster using the `multi` binary

Reviewer 3: ```suggestion
.Example output using the `multi` payload
```

Reviewer 5: I think whether it's:
```suggestion
.Example output of the supported archtecture using the `multi` installer binary
```

[source,terminal]
----
release@sha256:27d41f99e0784a2f596e773799983147b794df48d1f7536d51a95a60945b1931 -o jsonpath="{ .metadata.metadata}"
{"release.openshift.io/architecture":"multi"}
----
+

> With the `multi` installer binary, your cluster supports `arm64`, `amd64`, `s390x`, and `ppc64le`. Otherwise, you can only install the architecture defined in the `release architecture` output from the `./openshift-install version` command. 

Reviewer 3: ```suggestion
With the `multi` installer binary, your cluster supports `arm64`, `amd64`, `s390x`, and `ppc64le`. Otherwise, you can only install the architecture defined in the `release architecture` output from the `openshift-install version` command. 
```

> With the `multi` installer binary, your cluster supports `arm64`, `amd64`, `s390x`, and `ppc64le`. Otherwise, you can only install the architecture defined in the `release architecture` output from the `./openshift-install version` command. 

Reviewer 3: ```suggestion
With the `multi` installer binary, your cluster supports `arm64`, `amd64`, `s390x`, and `ppc64le`. Otherwise, you can only install the architecture defined in the `release architecture` output from the `openshift-install version` command. 
```
